azurerm/utils/type.go

```diff
@@ -0,0 +1,67 @@
+package utils
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+func ParseFloat(value string) (*float64, error) {
+	if v, err := strconv.ParseFloat(value, 64); err == nil {
+		return &v, nil
+	}
+	return nil, fmt.Errorf("Value %q is not a float number", value)
+}
+
+func ParseFloats(values []string) ([]float64, error) {
+	numbers := []float64{}
+	for _, v := range values {
+		if f, err := ParseFloat(v); err == nil {
+			numbers = append(numbers, *f)
+		} else {
+			return nil, err
+		}
+	}
+	return numbers, nil
+}
+
+func ParseBool(value string) (*bool, error) {
+	if v, err := strconv.ParseBool(value); err == nil {
+		return &v, nil
+	}
+	return nil, fmt.Errorf("Value %q is not a boolean", value)
+}
+
+func FormatBool(b bool) string {
+	return strconv.FormatBool(b)
+}
+
+func FormatFloat(f float64) string {
+	return strconv.FormatFloat(f, 'f', -1, 64)
+}
+
+func FormatFloats(values []float64) []string {
+	numbers := []string{}
+	for _, f := range values {
+		number := FormatFloat(f)
+		numbers = append(numbers, number)
+	}
+	return numbers
+}
+
+func TypeRelaxedEquals(old string, new string) bool {
+	if strings.EqualFold(old, new) {
```

Comment:
It's really been a long time since I've implemented this. But as far as I remember it happened that the advanced_filter should be re-created because the state apparently deviated from the target state. The problem was only solved with this softened validation.